src/Voron/Data/BTrees/Tree.MultiTree.cs

```diff
@@ -173,6 +359,49 @@ private void ExpandMultiTreeNestedPageSize(Slice key, Slice value, byte* nestedP
             }
         }
 
+        private void MultiBulkAddOnNewValue(Slice key, ReadOnlySpan<Slice> values, int maxNodeSize)
+        {
+            var requiredPageSize = Constants.Tree.PageHeaderSize + Constants.Tree.NodeHeaderSize;
+            for (int i = 0; i < values.Length; i++)
+            {
+                requiredPageSize += TreeSizeOf.NodeEntry(-1, values[i], 0) + Constants.Tree.NodeHeaderSize;
+            }
+            
+            if (requiredPageSize > maxNodeSize)
+            {
+                var tree = Create(_llt, _tx, key, TreeFlags.MultiValue);
+                _tx.AddMultiValueTree(this, key, tree);
+                DirectAdd(key, sizeof(TreeRootHeader), TreeNodeFlags.MultiValuePageRef,out _).Dispose();
+                
+                foreach (var value in values)
+                    tree.AddKeyOnly(value).Dispose();
```

Comment:
Maybe not for this PR but it would make sense to see if we can build a `MultiDirectAdd` operation that takes `ReadOnlySpan<Slice>` and exploits the fact that we know we are doing multiple adds in 1 go.